core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -9633,6 +9641,32 @@ private boolean isUserStoreChainNeeded(List<String> userStorePreferenceOrder) th
     private boolean generateUserStoreChain(String userName, Object credential, boolean domainProvided,
                                            List<String> userStorePreferenceOrder) throws UserStoreException {
 
+        // If domain name is provided, directly authenticate using the corresponding user store.
+        if (domainProvided) {
+            String providedDomainName = this.getUserStore(userName).getDomainName();
```

Comment:
Why don't we use `UserCoreUtil.extractDomainFromName(userName)` here.

Reply:
Addressed in https://github.com/wso2/carbon-kernel/pull/2765